invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/search/SearchResultItem.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2022 CERN.
+ *
+ * Invenio is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+import PropTypes from "prop-types";
+import React, { Component } from "react";
+import _get from "lodash/get";
+import { Table } from "semantic-ui-react";
+import isEmpty from "lodash/isEmpty";
+import { withState } from "react-searchkit";
+import { Actions } from "@js/invenio_administration/actions/Actions.js";
+import { AdminUIRoutes } from "@js/invenio_administration/routes.js";
+
+class SearchResultItemComponent extends Component {
+  refreshAfterAction = () => {
+    const { updateQueryState, currentQueryState } = this.props;
+    updateQueryState(currentQueryState);
+  };
+
+  createdBySystem = () => {
+    const { result } = this.props;
+    return result.system_created;
+  };
+
+  render() {
+    const {
+      title,
+      resourceName,
+      result,
+      columns,
+      displayEdit,
+      displayDelete,
+      actions,
+      apiEndpoint,
+      idKeyPath,
+      listUIEndpoint,
+    } = this.props;
+
+    const resourceHasActions =
+      displayEdit || displayDelete || !isEmpty(actions);
+    return (
+      <Table.Row>
+        {columns.map(([property, { text, order }], index) => {
+          return (
+            <Table.Cell key={`${text}-${order}`} data-label={text}>
+              {index === 0 && (
+                <a
+                  href={AdminUIRoutes.detailsView(
+                    listUIEndpoint,
+                    result,
+                    idKeyPath
+                  )}
+                >
+                  {_get(result, property)}
+                </a>
+              )}
+              {index !== 0 && _get(result, property)}
+            </Table.Cell>
+          );
+        })}
+        {resourceHasActions && (
+          <Table.Cell>
+            <Actions
+              title={title}
+              resourceName={resourceName}
+              apiEndpoint={apiEndpoint}
+              displayEdit={displayEdit}
+              disableEdit={this.createdBySystem()}
+              displayDelete={displayDelete}
+              disableDelete={this.createdBySystem()}
```

Comment:
Should a super admin be able to do that? I am thinking about support... we might want to be able to easily change the set from the web interface.
Otherwise, we need to create a recipe to do it programmatically.

Reply:
I am not sure about this, if a set is created automatically when we create a community, I personally think an admin should not be able to change it mainly because (_i_) the spec is unique and cannot be updated in the OAI set, (_ii_) the search pattern "should" never change because it's pointing to the community identifier which cannot be updated either. The only things left to update are description or name, based on this, my humble opinion is that it's safe enough to have a small recipe to update the OAI sets if needed. Feel free to challenge this, might be that I am missing something!